Sources/PointFree/SiteMiddleware.swift

```diff
@@ -82,8 +84,12 @@ public func siteMiddleware(
 
   return await withDependencies {
     $0.currentUser = currentUser
-    $0.requestID = requestID
     $0.currentRoute = siteRoute ?? .home
+    $0.episodeProgresses = .init(
+      progresses.map { ($0.episodeSequence, $0) },
+      uniquingKeysWith: { $1 }
+    )
+    $0.requestID = requestID
```

Comment:
Since this actually only ended up being used on the homepage, I could be convinced to pull this out of dependencies and just thread it through to the view. But, in the future I'm sure we will want to show this info in more places, and at that time we will probably want to do this. Just a thought.